File: data/data_original/at_coder/agc013/A/4588709.py
N = int(input())
AN = list(map(int,input().split()))

ans = 1		#??????1
beforeDiff = 0

for i in range(1,N):
	diff = AN[i] - AN[i-1]
	
	if beforeDiff==0:	#?????
		beforeDiff = diff
	elif beforeDiff*diff < 0:	#????????????
		beforeDiff = 0	#??1????????????diff????????0
		ans += 1
				#???????????????
print(ans)

File: storage/data_transpiled/at_coder/agc013/A/4588709.java
public static int N = Integer . parseInt ( input ) {
  int [ ] AN = Lists . newArrayList ( ) ;
  for ( int i = 0 ;
  i < N ;
  i ++ ) {
    AN [ i ] = Integer . parseInt ( input . nextLine ( ) ) ;
  }
  int ans = 1 ;
  int beforeDiff = 0 ;
  for ( int i = 1 ;
  i < N ;
  i ++ ) {
    int diff = AN [ i ] - AN [ i - 1 ] ;
    if ( beforeDiff == 0 ) {
      beforeDiff = diff ;
    }
    else if ( beforeDiff * diff < 0 ) {
      beforeDiff = 0 ;
      ans ++ ;
    }
  }
  return ans ;
}
